DataFormats/Provenance/interface/ProductRegistry.h

```diff
@@ -163,6 +143,30 @@ namespace edm {
       AliasToOriginalVector aliasToOriginal_;
     };
 
+  private:
+    //The following three routines are only called by SignallingProductRegistryFiller
+    void addProduct_(ProductDescription const& productdesc);
+
+    ProductDescription const& addLabelAlias_(ProductDescription const& productdesc,
+                                             std::string const& labelAlias,
+                                             std::string const& instanceAlias);
+
+    // triggers callbacks for modules watching registration
+    template <typename F>
+    void addFromInput_(edm::ProductRegistry const& iReg, F const& iCallback) {
```

Comment:
If `const`-only access is not important, this could use the universal reference for `F`
```suggestion
    template <typename F>
    void addFromInput_(edm::ProductRegistry const& iReg, F&& iCallback) {
```
Universal reference would allow mutable functors.

Reply:
done